src/main/java/org/openrewrite/analysis/trait/expr/MethodAccess.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.analysis.trait.expr;
+
+import fj.data.Validation;
+import lombok.AllArgsConstructor;
+import org.openrewrite.Cursor;
+import org.openrewrite.analysis.InvocationMatcher;
+import org.openrewrite.analysis.trait.TraitFactory;
+import org.openrewrite.analysis.trait.util.TraitErrors;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+public interface MethodAccess extends Expr, Call {
+    String getSimpleName();
+
+    List<Expression> getArguments();
+
+    boolean matches(InvocationMatcher callMatcher);
+
+    boolean matches(MethodMatcher methodMatcher);
```

Comment:
I would suggest not accepting `MethodMatcher`. Only accept `InvocationMatcher` and force API consumers to do the conversion